src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/BuildPropertyPage/DefineConstantsCSharpValueProvider.cs

```diff
@@ -51,7 +56,11 @@ public override async Task<string> OnGetUnevaluatedPropertyValueAsync(string pro
     // 2. to override IsValueDefinedInContextAsync, as this will always return false
     private async Task<string?> GetUnevaluatedDefineConstantsPropertyValueAsync()
     {
-        await ((ConfiguredProject2)_project).EnsureProjectEvaluatedAsync();
+        if (_project is ConfiguredProject2 configuredProject2)
+        {
+            await configuredProject2.EnsureProjectEvaluatedAsync();
+        }
```

Comment:
Is there a code path where `_project` is not of the expected type? If so I can imagine that causing problems if we don't consistently wait for evaluation to complete.

Reply:
Yes, but only in tests

Reply:
Can we improve the test then? I assume we're using Moq, which supports creating a mock that implements multiple interfaces.

Reply:
Done, I created a composite test type implementing both `ConfiguredProject` and `ConfiguredProject2`